plugins/inputs/bond/bond.go

```diff
@@ -118,10 +155,100 @@ func (bond *Bond) gatherBondPart(bondName string, rawFile string, acc telegraf.A
 	return fmt.Errorf("Couldn't find status info for '%s' ", bondName)
 }
 
+func (bond *Bond) readSysFiles(bondDir string) (sysFiles, error) {
+	/*
+		Files we may need
+		bonding/mode
+		bonding/slaves
+		bonding/ad_num_ports
+	*/
+	var err error
+	var output sysFiles
+
+	file, err := os.ReadFile(bondDir + "/bonding/mode")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting '%s' interface: %v", bondDir+"/bonding/mode", err)
+	}
+	output.ModeFile = strings.TrimSpace(string(file))
+	file, err = os.ReadFile(bondDir + "/bonding/slaves")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting '%s' interface: %v", bondDir+"/bonding/slaves", err)
+	}
+	output.SlaveFile = strings.TrimSpace(string(file))
+	if bond.BondType == "IEEE 802.3ad Dynamic link aggregation" {
+		file, err = os.ReadFile(bondDir + "/bonding/ad_num_ports")
+		if err != nil {
+			return sysFiles{}, fmt.Errorf("error inspecting '%s' interface: %v", bondDir+"/bonding/ad_num_ports", err)
+		}
+		output.ADPortsFile = strings.TrimSpace(string(file))
+	}
+	return output, nil
+}
+
+func (bond *Bond) gatherSysDetails(bondName string, files sysFiles, acc telegraf.Accumulator) error {
```

Comment:
AFAICS this function only returns `nil` and never a real error, so please remove the return variable
```suggestion
func (bond *Bond) gatherSysDetails(bondName string, files sysFiles, acc telegraf.Accumulator) {
```
to make this obvious to the caller.